### ISSUE TITLE
Checksum for files

### ISSUE THREAD

**Issue by facelessuser** [facelessuser/Rummage#166]
I already have code that could do this, would probably just need to massage the API to work better for this, but this could be kind of cool.  Probably won't hash together multiple files or anything, but just operate on a single file.

This is more an experimental idea.  I'd love to tap into the OSs file manager and just bring up its context menu, but that seems like more work than I care to do or have time for. But I think this could work as I've already done the work for another project.  I'd just be reusing it here.

**Comment by facelessuser** [comment]
I actually have this working on macOS, but unfortunately wxPython is inconsistent in the way it handles the menus we dynamically generate.  Windows seems to never execute callbacks for the checksum sub menu, while macOS can do it all day long.

Ugh, it's these platform differences that exhaust me.

**Comment by facelessuser** [comment]
So it turns out that my understanding of menus was wrong:

> All menus must be created on the heap because all menus attached to a menubar or to another menu will be deleted by their parent when it is deleted. The only exception to this rule are the popup menus (i.e. menus used with wx.Window.PopupMenu ) as wxWidgets does not destroy them to allow reusing the same menu more than once. But the exception applies only to the menus themselves and not to any submenus of popup menus which are still destroyed by wxWidgets as usual and so must be heap-allocated. As the frame menubar is deleted by the frame itself, it means that normally all menus used are deleted automatically.

This makes sense, and refactoring accordingly makes all the difference.

**Comment by facelessuser** [comment]
Need to show dialog with current progress (for big files), and an option to cancel/abort current processing.  Dialog should have a text box with the hash for easy copy and paste.

**Comment by facelessuser** [comment]
Turns out there was a difference between windows and others.  When building the dynamic window and using `PopupMenu` to show it, submenus  for windows needs to bind events to the main menu, whereas others need you to bind the events to the main menu.  I had assumed, based on the description in the docs, that the submenu was getting destroyed immediately, and that is why we had to bind the menu to the main parent.  Turns out, it is just inconsistent behavior across platforms...
